classes/asserters/request/url.php

```diff
@@ -28,4 +70,244 @@ public function setParentAsserter(atoum\http\asserters\request $request)
 
 		return $this->setWith($this->request->getValue()->getUrl());
 	}
+
+    public function setSchemeAsserterFactory(\closure $factory = null)
+    {
+        $factory = $factory ?: function(asserter\generator $generator) {
+            return new url\scheme($generator);
+        };
+
+        $generator = $this->generator;
+        $test = & $this->test;
```

Comment:
Yes it is : `$this->test` is not directly set, if I only capture a value (which will be the case because `this->test` is null) the closure won't get the correct reference to `$this->test` when it gets set.